src/analysis/special_functions/log.lean

```diff
@@ -172,6 +172,32 @@ begin
   { rintro (rfl|rfl|rfl); simp only [log_one, log_zero, log_neg_eq_log], }
 end
 
+lemma log_le_sub_one_of_pos {x : ℝ} (hx : 0 < x) : log x ≤ x - 1 :=
+begin
+  have h := add_one_le_exp (log x),
+  rw exp_log hx at h,
+  rw le_sub_iff_add_le,
+  exact h,
+end
+
+lemma log_div_self_antitone_on : antitone_on (λ x : ℝ, log x / x) {x | exp 1 ≤ x} :=
+begin
+  rw antitone_on,
+  intros x hex y hey hxy,
+  have x_pos : 0 < x := (exp_pos 1).trans_le hex,
+  have y_pos : 0 < y := (exp_pos 1).trans_le hey,
+  have hlogx : 1 ≤ log x,
+  { rwa le_log_iff_exp_le x_pos, },
+  have hyx : 0 ≤ y / x - 1,
+  { rwa [le_sub_iff_add_le, le_div_iff x_pos, zero_add, one_mul], },
+  rw [div_le_iff y_pos, ←sub_le_sub_iff_right (log x)],
+  calc
+    log y - log x = log (y / x)           : by rw [log_div (y_pos.ne') (x_pos.ne')]
+    ...           ≤ (y / x) - 1           : log_le_sub_one_of_pos (div_pos y_pos x_pos)
+    ...           ≤ log x * (y / x - 1)   : le_mul_of_one_le_left hyx hlogx
+    ...           = log x / x * y - log x : by ring,
+end
```

Comment:
If you replace the first line with `simp only [antitone_on, mem_set_of_eq]`, it makes the hypotheses a bit nicer.

Also, for one-line tactic proofs with `have`, you might consider `have foo : bar := by baz` to get the proofs more out of the way.

```suggestion
begin
  simp only [antitone_on, mem_set_of_eq],
  intros x hex y hey hxy,
  have x_pos : 0 < x := (exp_pos 1).trans_le hex,
  have y_pos : 0 < y := (exp_pos 1).trans_le hey,
  have hlogx : 1 ≤ log x := by rwa le_log_iff_exp_le x_pos,
  have hyx : 0 ≤ y / x - 1 := by rwa [le_sub_iff_add_le, le_div_iff x_pos, zero_add, one_mul],
  rw [div_le_iff y_pos, ←sub_le_sub_iff_right (log x)],
  calc
    log y - log x = log (y / x)           : by rw [log_div (y_pos.ne') (x_pos.ne')]
    ...           ≤ (y / x) - 1           : log_le_sub_one_of_pos (div_pos y_pos x_pos)
    ...           ≤ log x * (y / x - 1)   : le_mul_of_one_le_left hyx hlogx
    ...           = log x / x * y - log x : by ring,
end
```

Reply:
I think the `:= by` looks really weird, you can simply write `, by`.

Reply:
But at that point, why even write the `by`?

Reply:
@BoltonBailey `by` causes Lean to issue an error if the tactic following it does not close the goal. It's like putting `{` and `}` around it.

I like `:= by`, even if it might look weird, because it helps avoid breaking the soft rule that there shouldn't be more than one comma-separated tactic per line, but that's just my reasoning.

Reply:
Indeed, `{ ... }` was how it started, I just put the new block on the next line. It would be nice if the style guide gave guidance on which of these four permutations is preferred. Until then, I'm just going to put this back to awaiting-review.